File: Shared/TerminologyUtilities.vb
'*********************************************************************************************
' STSimStockFlow: A SyncroSim Module for the ST-Sim Stocks and Flows Add-In.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************

Imports SyncroSim.Core

Module TerminologyUtilities

    Public Function GetTerminology(
        ByVal terminologySheet As DataSheet,
        ByVal columnName As String) As String

        Dim dr As DataRow = terminologySheet.GetDataRow()

        If (dr Is Nothing OrElse dr(columnName) Is DBNull.Value) Then
            Return "Units"
        Else
            Return CStr(dr(columnName))
        End If

    End Function

    Public Function GetTimestepUnits(ByVal project As Project) As String

        Dim dr As DataRow = project.GetDataSheet("STSim_Terminology").GetDataRow()

        If (dr Is Nothing OrElse dr("TimestepUnits") Is DBNull.Value) Then
            Return "Timestep"
        Else
            Return CStr(dr("TimestepUnits"))
        End If

    End Function

    Public Sub GetStratumLabelStrings(
        ByVal terminologyDataSheet As DataSheet,
        ByRef primaryStratumLabel As String,
        ByRef secondaryStratumLabel As String)

        Dim dr As DataRow = terminologyDataSheet.GetDataRow()

        primaryStratumLabel = "Stratum"
        secondaryStratumLabel = "Secondary Stratum"

        If (dr IsNot Nothing) Then

            If (dr("PrimaryStratumLabel") IsNot DBNull.Value) Then
                primaryStratumLabel = CStr(dr("PrimaryStratumLabel"))
            End If

            If (dr("SecondaryStratumLabel") IsNot DBNull.Value) Then
                secondaryStratumLabel = CStr(dr("SecondaryStratumLabel"))
            End If

        End If

    End Sub

End Module

File: Shared/TerminologyUtilities.cs
// STSimStockFlow: A SyncroSim Module for the ST-Sim Stocks and Flows Add-In.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System;
using System.Data;
using SyncroSim.Core;

namespace SyncroSim.STSimStockFlow
{
	internal static class TerminologyUtilities
	{
		public static string GetTerminology(DataSheet terminologySheet, string columnName)
		{

			DataRow dr = terminologySheet.GetDataRow();

			if (dr == null || dr[columnName] == DBNull.Value)
			{
				return "Units";
			}
			else
			{
				return Convert.ToString(dr[columnName]);
			}
		}

		public static string GetTimestepUnits(Project project)
		{
			DataRow dr = project.GetDataSheet("STSim_Terminology").GetDataRow();

			if (dr == null || dr["TimestepUnits"] == DBNull.Value)
			{
				return "Timestep";
			}
			else
			{
				return Convert.ToString(dr["TimestepUnits"]);
			}
		}

		public static void GetStratumLabelStrings(DataSheet terminologyDataSheet, ref string primaryStratumLabel, ref string secondaryStratumLabel, ref string tertiaryStratumLabel)
		{
			DataRow dr = terminologyDataSheet.GetDataRow();

			primaryStratumLabel = "Stratum";
			secondaryStratumLabel = "Secondary Stratum";
			tertiaryStratumLabel = "Tertiary Stratum";

			if (dr != null)
			{
				if (dr["PrimaryStratumLabel"] != DBNull.Value)
				{
					primaryStratumLabel = Convert.ToString(dr["PrimaryStratumLabel"]);
				}

				if (dr["SecondaryStratumLabel"] != DBNull.Value)
				{
					secondaryStratumLabel = Convert.ToString(dr["SecondaryStratumLabel"]);
				}

				if (dr["TertiaryStratumLabel"] != DBNull.Value)
				{
					tertiaryStratumLabel = Convert.ToString(dr["TertiaryStratumLabel"]);
				}
			}
		}
	}
}